src/main/dotcode.c

```diff
@@ -1698,14 +1698,19 @@ SEXP attribute_hidden do_dotCode(SEXP call, SEXP op, SEXP args, SEXP env)
 			    type2char(t), na + 1);
 	    /* Used read-only, so this is safe */
 #ifdef USE_RINTERNALS
-	    cargs[na] = (void*) DATAPTR(s);
+            if (!ALTREP(s))
+                cargs[na] = (void*) DATAPTR(s);
+            else {
 #else
-	    n = XLENGTH(s);
-	    SEXP *lptr = (SEXP *) R_alloc(n, sizeof(SEXP));
-	    for (R_xlen_t i = 0 ; i < n ; i++) lptr[i] = VECTOR_ELT(s, i);
-	    cargs[na] = (void*) lptr;
+                n = XLENGTH(s);
```

Comment:
Not sure how much this matters to R-core, but you are changing from tabs to spaces here and a few more places

Reply:
yeah, emacs settings. hopefully does not matter, because spaces are better.... :)